packages/components/src/keyboard-shortcuts/index.js

```diff
@@ -2,7 +2,10 @@
  * External dependencies
  */
 import Mousetrap from 'mousetrap';
-import 'mousetrap/plugins/global-bind/mousetrap-global-bind';
+if ( typeof window !== 'undefined' ) {
+	// only initialize the global bind if we're in a browser environment
+	require( 'mousetrap/plugins/global-bind/mousetrap-global-bind' );
```

Comment:
I hate this, but I'm unsure how else to handle it. Mousetrap returns early when `window` isn't present, but the plugin has no such protection.

Reply:
Should probably fixed upstream?

Reply:
> Should probably be fixed upstream?

If there was an issue, I would suggest so.

Looking at [the code](https://github.com/ccampbell/mousetrap/blob/master/plugins/global-bind/mousetrap-global-bind.js), it's not immediately obvious to me what the issue is.

There's an alternative that we copy and adapt the code from the plugin directly in this file.

I don't have a strong opinion on approach, even as implemented here. Would be good to have a better understanding of the problem this is addressing though.

Reply:
Worth starting by opening an issue at upstream? https://github.com/ccampbell/mousetrap/issues

Reply:
> > Should probably be fixed upstream?
> 
> If there was an issue, I would suggest so.
> 
> Looking at [the code](https://github.com/ccampbell/mousetrap/blob/master/plugins/global-bind/mousetrap-global-bind.js), it's not immediately obvious to me what the issue is.
> 
> There's an alternative that we copy and adapt the code from the plugin directly in this file.
> 
> I don't have a strong opinion on approach, even as implemented here. Would be good to have a better understanding of the problem this is addressing though.

My (admittedly blunt) take is that if it's available as an npm, it should at least not break things when used on the server side (read: node) but fall back to a no-op.

Reply:
And as Ben points out, `mousetrap` itself takes precautions to prevent that from happening -- it's just the `global-bind` plugin that doesn't.

Reply:
Precautions against what? I see no references to browser globals in the `global-bind` plugin code.

Reply:
The plugin goes after the `Mousetrap` global and the `Mousetrap.prototype`, but `mousetrap` returns early when `window` is not present and never sets the `Mousetrap` global. So it's not a browser global that's missing, it's `Mousetrap`.

Reply:
filed https://github.com/ccampbell/mousetrap/pull/465 upstream